src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -424,6 +426,8 @@ protected internal virtual void OnProjectRemoved(ProjectId projectId)
                 var newSolution = this.SetCurrentSolution(oldSolution.RemoveProject(projectId));
 
                 this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectRemoved, oldSolution, newSolution, projectId);
+
+                _isProjectUnloading = false;
```

Comment:
+1 on the finally.  If only because someone might put a return in this code in the future and completely f-up the system.